cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -486,22 +510,10 @@ public Void call() throws Exception {
           // TODO this should be done in the output committer, to make the M/R fail if addPartition fails
           boolean success = succeeded;
           for (Map.Entry<String, Dataset> dsEntry : context.getOutputDatasets().entrySet()) {
-            String datasetName = dsEntry.getKey();
-            Dataset outputDataset = dsEntry.getValue();
-
-            if (outputDataset instanceof DatasetOutputCommitter) {
-              try {
-                if (succeeded) {
-                  ((DatasetOutputCommitter) outputDataset).onSuccess();
-                } else {
-                  ((DatasetOutputCommitter) outputDataset).onFailure();
-                }
-              } catch (Throwable t) {
-                LOG.error(String.format("Error from %s method of output dataset %s.",
-                                        succeeded ? "onSuccess" : "onFailure", datasetName), t);
-                success = false;
-              }
-            }
+            success = success && commitOutput(succeeded, dsEntry.getKey(), dsEntry.getValue());
```

Comment:
The previous behavior was that if one of the committers failed, then the remainder would still attempt to commit.
That behavior is no longer true. Is that intended?